File: 1 basic/cpp/1032. 挖掘机技术哪家强(20).cpp
#include<cstdio>

const int N = 1e5+5;
int a[N];

int main(){
	int n;
	int id,score;
	scanf("%d",&n);
	for(int i=1;i<=n;++i){
		scanf("%d%d",&id,&score);
		a[id] += score;
	}
	score = id = 0;
	for(int i=1;i<=n;++i){
		if(a[i] > score){
			id = i;
			score = a[i];
		}
	}
	printf("%d %d\n",id,score);
	
	
	return 0;
}

File: 1 basic/py/1032. 挖掘机技术哪家强(20).py
N = int(input())
d = {}
Max_id, Max_score = 1,0
for i in range(N ):
    id,score = list( map(int, input().split() ) )
    tmp = d[id] = d.get(id, 0) + score
    if Max_score < tmp:
        Max_score = tmp
        Max_id = id
print(Max_id, Max_score)